### ISSUE TITLE
Ignore indent on comment node to allow 0 indentation comments in jsx.

### ISSUE THREAD

**Issue by jhicken** [yannickcr/eslint-plugin-react#1781]
Sometimes I comment features out temporarily. And I like to set the indentation to be at 0 so that it is more visible to people working on the code. In the middle of a large tree of jsx nodes this really helps.

**This is what it looks like.**
```
<div>
  <div>
{/*
      comment
*/}
  </div>
</div>
```

**These are work arounds.**

valid but sad
```
<div>
  <div>
    {/*
      comment
    */}
  </div>
</div>
```

also valid but sad
```
<div>
  <div>
    {/* eslint-disable-next-line */}
{/*
    comment
*/}
  </div>
</div>
```

I could probably write a custom rule for this but it seems like something that might could be implemented in this project.

**Comment by ljharb** [comment]
```jsx
<div>
  <div>
{/* eslint-disable-line react/jsx-indent */ /*
      comment
*/}
  </div>
</div>
```
?

Either way, that would require a special jsx-indent option for comments - and that's something i wouldn't want to add unless the core indent rule had the same option.

**Comment by jhicken** [comment]
@ljharb I'm not sure if that was a question or an answer or a possible solution. 

That code you pasted does not currently work. Are you suggesting a change to make that work?

Why would the core indent rule need a special option?

For non jsx.. you can just use `ignoreComments`. And it works fine. There is no equivalent for jsx comments due to a comment actually being a child node evaulation. If jsx comments payed attention to `ignoreComments` for indentation and said ignore the parent node indentation.... That would work for me too.

**Comment by ljharb** [comment]
@jhicken i was on mobile, so I couldn't try it myself, but it was a suggestion to see if it would work.

What I meant was, I think `react/jsx-indent` should not add any options that `indent` doesn't have, specifically so that both rules can format code in the same ways.

I'm not sure what could possibly work here, due to the issues you're pointing out. Is "differentiate with syntax highlighting, not indentation" not an option?

**Comment by jhicken** [comment]
If I understand correctly your asking if I can accomplish this with syntax highlighting. If thats whats your asking no I cannot.. due to my linter being connected to my build process. So when linting fails the build fails.

I could make some sort of syntax highlighter and force the comments to start and end at the current indentation. But I not everybody on my team uses the same editors. So thats probably not a good option. And If I where to do that I would just prefer to create my own eslint plugin. 

Has the `ignoreComments` ignoring a jsx comments parent thing that i described been explored?

**Comment by ljharb** [comment]
I’d strongly recommend not tying linting into the build - they’re two unrelated concerns.

I don’t think it’s been explored. If you want to give it a shot, I’m happy to review.

**Comment by jhicken** [comment]
I disagree with your first comment. They are two concerns... and they are closely related. If code quality does not meet team standards then it should not continue to be built in the CI process to go to production.

That doesn't really matter though. Different teams will do what works for them.

I may give it a look.. but it wont be for a few weeks. busy busy.

**Comment by ljharb** [comment]
Oh I totally agree that the linter should gate any merging, and certainly any deploying - but for local dev that's not a concern, and the build should be testable even before the linter passes :-)

Sounds good, I'll review whenever you have time.
